### DISCUSSION TITLE
Cython compatibility with pytest

### DISCUSSION THREAD

**Discussion by emiWilson** [pytest-dev/pytest#9729] [Q&A]
How can I use pytest to test cython functions that are defined as cdefs (not cpdefs like in the documentation: https://pytestguide.readthedocs.io/en/latest/pytestGuide/)?

**Comment by RonnyPfannschmidt** [comment]
Any function you don't expose to python, you cannot test with python

On the c side of things people tend to use cffi or ctypes

Personally i would recommend that you have a a private cpdef based wrapper for the cdef to facilitate invocation in tests

It's OK to put such wrappers in private modules
